libraries/node-core-library/src/Sort.ts

```diff
@@ -231,4 +231,79 @@ export class Sort {
       set.add(item);
     }
   }
+
+  /**
+   * Sort the keys given in an object
+   *
+   * @example
+   *
+   * ```ts
+   * console.log(Sort.sortKeys({ c: 3, b: 2, a: 1 })); // { a: 1, b: 2, c: 3}
+   * ```
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public static sortKeys<T extends Partial<Record<string, any>> | Array<any>>(
+    object: T,
+    { deep, compare }: { deep?: boolean; compare?: (x: string, y: string) => number } = {
+      deep: false,
+      compare: Sort.compareByValue
+    }
+  ): T {
+    function isPlainObject(obj: unknown): boolean {
+      return obj !== null && typeof obj === 'object';
+    }
+    if (!isPlainObject(object) && !Array.isArray(object)) {
+      throw new TypeError(`Expected object or array`);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const cache: WeakMap<T, any> = new WeakMap<T, any>();
```

Comment:
It looks like the `WeakMap` value is always either an object or else an array.  If it cannot ever be `undefined`, then the lookup speed will be twice as fast if we replace this:
```ts
      if (cache.has(arr as T)) {
        return cache.get(arr as T)!;
      }
```
...with this:

```ts
      const value: unknown | undefined = cache.get(arr as T)
      if (value !== undefined) {
        return value;
      }
```
... and also the `!` is no longer needed for `get(arr as T)!`.